src/bucket/Bucket.h

```diff
@@ -42,7 +43,29 @@ class Bucket : public std::enable_shared_from_this<Bucket>,
   public:
     // Helper class that reads through the entries in a bucket, used internally
     // during merging.
-    class InputIterator;
+    class InputIterator
```

Comment:
couple things:
* in general it's better to have a separate commit for code moves like this ("moved Bucket::InputIterator definition from cpp to h file")
* in this case, I don't see the reason to move it to the .h file, it just pollutes the .h file (it's only used internally), if all you wanted was to stop having inline declarations, you can still achieve the same thing by having both the class definition and implementation in the same file

Reply:
nm, I didn't see the rest of the changes in the file - was confused by the comment that is no longer relevant.

Reply:
now that you are making this available outside of this file, it's the occasion to remove the code in `BucketApplicator` that does the same thing from what I can tell